### ISSUE TITLE
INSPIRE Workshop materials

### ISSUE THREAD

**Issue by danielnavarrogeo** [INSPIRE-MIF/helpdesk-validator#123]
Please replace these lines with your question or discussion topic.
# Practical examples for INSPIRE validator
INSPIRE Workshop, 1-2 october 2019, Ispra, Italy
## Metadata 2.0 - Dataset
Common Requirements 
* [md dataset 1](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_01_metadata_2_0_dataset.xml)
* [md dataset 2](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_02_metadata_2_0_dataset.xml)

Conformance Class 1 
* [md dataset 3](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_03_metadata_2_0_dataset.xml)

Conformance Class 2
* [md dataset 4](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_04_metadata_2_0_dataset.xml)

## Metadata 2.0 - Service
Conformance Class 3 
* [md service 5](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_05_metadata_2_0_service.xml)

Conformance class 6 
* [md service 6](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_06_metadata_2_0_service.xml)

## View service
* [vs wms 7](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_07_service_wms.xml)

* [vs wms 8](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_08_service_wms.xml)

* [vs wms 9](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_09_service_wms.xml)

* [vs wms 10](https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_10_service_wms.xml)

**Comment by bor8** [comment]
https://etherpad.wikimedia.org/p/ETF

**Comment by ghost** [comment]
The link provided above can be used to provide Code-Snippets for step-by-step instructions in order to validate SDI-Ressources in ETF-Validator using API.

**Comment by ghost** [comment]
Could you please create step-by-step instructions for validation of single and multiple metadata records using API including Code-Snippets?

**Comment by ghost** [comment]
> 
> 
> Could you please create step-by-step instructions for validation of single and multiple metadata records using API including Code-Snippets?

Step-by-step instructions are partly documented here:
http://docs.etf-validator.net/v2.0/Developer_manuals/WEB-API.html#_overview

**Comment by ghost** [comment]
Step-by-step instruction for customising BaseX based tests (Exstension of INSPIRE ETS):

1. In order to understand the structure of the ETS open https://github.com/inspire-eu-validation/ets-repository/blob/master/metadata/2.0/common/ets-md-common-bsxets.xml 
2. Following elements (XPath) should be considered for customizing:

- /ExecutableTestSuite/tags/tag/@ref is used to reference the XML which is defining one specific category in INSPIRE Validator

![grafik](https://user-images.githubusercontent.com/38787131/66030901-a6a46e00-e502-11e9-8f8b-f21fb78ad9ef.png)
This XML can be found using the ID provided by XPath to search the repository:
https://github.com/inspire-eu-validation/ets-repository/blob/master/include-metadata/Tag-EIDc6567beb-fc33-4f2e-865d-0c3ee5b3d1ae.xml
In this case the category is labeled with "Metadata (TG version 2.0) - BETA"

- /ExecutableTestSuite/testDriver/@ref is used to reference the Driver which will be used for validation. In total there are two drivers "BaseX" and "SoapUI".  IDs for both of them are fixed and can be copied from ETS. For BaseX based tests take any metadata related Conformance Class and for SoapUI use some OGC Service Conformance Class.

- /ExecutableTestSuite/translationTemplateBundle/@ref is used refernce the XML file which specifies exceptions for the messages provided in ETF-Validator. This file can be found using the ID from XPath:
https://github.com/inspire-eu-validation/ets-repository/blob/master/include-metadata/TranslationTemplateBundle-EID70a263c0-0ad7-42f2-9d4d-0d8a4ca71b52.xml

- /ExecutableTestSuite/supportedTestObjectTypes/testObjectType/@ref is used to reference TestObjectType. All IDs for TestObjectTypes can be found:

For SoapUI;
https://docs.etf-validator.net/v2.0/Developer_manuals/Developing_Executable_Test_Suites.html#soapui-test-object-types
For BaseX:
https://docs.etf-validator.net/v2.0/Developer_manuals/Developing_Executable_Test_Suites.html#basex-test-object-types

Basically there are two possibilities for customizing:
- Customizing of the ETS provided from INSPIRE
- Adding new ETS and used it together in combination with INSPIRE Tests

**Comment by ghost** [comment]
@danielnavarrogeo Could you please check if the documentation is fine?

**Comment by danielnavarrogeo** [comment]
Thank you @DeordD I think the documentation is fine.

**Comment by bor8** [comment]
@DeordD

Thanks for sharing.

You mentioned a mailing list or something like that where you can get information about INSPIRE or the ETF validator. Where can I find it?

@ All

Where and when do I get the workshop slides?

**Comment by MarcoMinghini** [comment]
> You mentioned a mailing list or something like that where you can get information about INSPIRE or the ETF validator. Where can I find it?

There are no mailing lists on the Validator. So I'm not sure what this means, @DeordD can you clarify?

> Where and when do I get the workshop slides?

As said during the workshop, as soon as we get the finalized version of the slides (with few changes based on the comments received during the workshop) and the step-by-step tutorials, we will create a new sub-folder in the community repository with all the materials. I will inform you here when this happens.

**Comment by ghost** [comment]
> You mentioned a mailing list or something like that where you can get information about INSPIRE or the ETF validator. Where can I find it?

@bor8 I suppose INSPIRE Community Forum is meant. You have section on tools there. Link to Forum:
https://inspire.ec.europa.eu/forum/tcc

**Comment by MarcoMinghini** [comment]
> I suppose INSPIRE Community Forum is meant.

Thanks for the hint! Yes, during the workshop I mentioned the INSPIRE Community Forum ('Software & Tools' section) as the place where we update the community on important news about the Validator (e.g. the release of a new version).

**Comment by MarcoMinghini** [comment]
Dear all,
I'm happy to announce that the workshop materials are finally available in this new folder in the Community space: https://github.com/inspire-eu-validation/community/tree/master/training%20material

Therefore, this issue will be soon closed unless there is some topics (relevant to the workshop) still to be discussed.

**Comment by ghost** [comment]
Dear Marco,

thanks a lot for publication of presentations and step-by-step tutorials! Very informative presentations and nice structure of the tutorials.

In the workshop it was stated that a validation of the keywords for INSPIRE Monitoring - Spatial scope and Priority Data Set should be available soon. In my opinion such validation is not yet provided.
 
When can we expect ATS and ETS on production instance in order to validate metadata records against the corresponding requirements?

Thanks in advance!

**Comment by ghost** [comment]
Dear @MarcoMinghini,

thanks a lot for your very quick response. 

Best regards,
Dejan

**Comment by MarcoMinghini** [comment]
One more question related to the workshop: among the priority improvements connected to the Validator, the 'effective usage of INSPIRE Validator API for massive tests' was highly rated.

Do you have any preference/hint for the language of a possible API client? At the moment there is no such API client for the ETF API, but one might be developed in the future.

Thank you,
Marco

**Comment by klimeto** [comment]
@MarcoMinghini 

Thanks for raising this question.

I have been using ETF API from Javascript and Python applications, so ETF API client libraries in both would be more than welcomed!

Thank you,
Tomas

**Comment by klimeto** [comment]
@MarcoMinghini 

The S3 URLs to sample metadata are not working:

https://s3-eu-west-1.amazonaws.com/geograma.public/inspire_workshop/example_01_metadata_2_0_dataset.xml

Results in:

```xml
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
<RequestId>EAED7FCE4C5D6BFD</RequestId>
<HostId>
CS2Gu3M71XTRN7dQx0rvgwOLTJAZR+xvaqVojljWnhtyaKzP+QzJ3NKrpp+qnwa0Ubblz6/KHOA=
</HostId>
</Error>
```

Are they meant to be persistent, or they were used only for the WS?

Many thanks,
Tomas

**Comment by MarcoMinghini** [comment]
@klimeto they were used for the workshop. However, we used them to create concrete step-by-step tutorials (on metadata validation) that you can find [here](https://github.com/inspire-eu-validation/community/tree/master/training%20material). We agreed to produce and publish this training material (as a concrete output of the workshop) so that it can benefit the whole community.
